### ISSUE TITLE
Error: import declarations may only appear at top level of a module

### ISSUE THREAD

**Issue by Francewhoa** [mzur/gnome-shell-wsmatrix#280]
## Intro

I created this ticket to facilitate communications about pull request https://github.com/mzur/gnome-shell-wsmatrix/pull/270. I'm assuming that one challenge per ticket.

## Steps to reproduce error

1. Replaced only the file `thumbnailsBox.js` with commit cb29bcc for gnome-shell-wsmatrix version 38

1. Using [Extension Manager](https://flathub.org/apps/com.mattjakeman.ExtensionManager), try activating gnome-shell-wsmatrix version 38 with commit cb29bcc This video shows this step at https://vk.com/video-225763421_456239018

1. The first challenge is that the extension can't be activated. The number 1 in this screenshot below shows this first challenge. 
![Screenshot-342](https://github.com/mzur/gnome-shell-wsmatrix/assets/119942/e8843925-3de5-46cd-8262-e4f441356c89)

1. The second challenge is when trying to activate the extension, this error message is display within **Extension Manager**  `import declarations may only appear at top level of a module`. The number 2 in this screenshot above shows error message. This video shows this error at https://vk.com/video-225763421_456239018

1. The needed end result is that users are able to activate the extension without error message. So that this other related challenge  is potentitally resolved about https://github.com/mzur/gnome-shell-wsmatrix/issues/274

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

## Used

• gnome-shell-wsmatrix 38 with commit cb29bcc
• Debian 12 Bookworm 64 bits
• GNOME 43.9
• [Extension Manager](https://flathub.org/apps/com.mattjakeman.ExtensionManager) 0.4.3
• Wayland

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

## What I tried that did not work

* Deactivated all GNOME extensions. Except gnome-shell-wsmatrix 38 with commit cb29bcc

* Fresh GNOME session

* Restart device

* Used gnome-shell-wsmatrix 38, replaced only the file `thumbnailsBox.js` with commit cb29bcc

* Replaced all files and all folders of gnome-shell-wsmatrix 38. Which included patched file `thumbnailsBox.js`.

**Comment by Francewhoa** [comment]
> Thanks! Unfortunately, I don't see the "import declarations" error message there. Maybe it is reported by some other process. You can search the whole journal by executing `journalctl`, then entering the `/` character, typing part of the error message (e.g. `import declarations`) and finally pressing Enter. This will search the journal for the error message (which could take a while).

Good morning @mzur :) 

I'll try to get this "import declarations" error message for you

On question for you below. The logs are very long. I'm trying to narrow down the location of the slippery-fish-elusive "import declarations" error message.

**Question**. In those steps to reproduce above at https://github.com/mzur/gnome-shell-wsmatrix/issues/280#issue-2275797879 When would this error message be likely be added to the logs? For example: 

* When I open [Extension Manager](https://flathub.org/apps/com.mattjakeman.ExtensionManager)?
* Or after patching `thumbnailsBox.js`, during the opening of a new GNOME session? 
* Or when else? 

In other words, when does GNOME validate those declarations?

**Comment by mzur** [comment]
The extension is loaded when GNOME Shell is started, i.e. after the login screen. You should be able to find the error if you search for the "import declarations" message text in the log. However, once I also encountered an extension error message that didn't show up in the usual gnome-shell log. I don't remember where else I found it, unfortunately. Maybe you could try searching all of `journalctl` (without argument) if you can't find the message in the gnome-shell log.

You could also try to change the multi-line import [here](https://github.com/mzur/gnome-shell-wsmatrix/commit/cb29bccc6663c4b37033a9e907b6821ad421043f#diff-70c1f2497f1d50d3fa22a5bab40e173e861f7ecb414cf792c2d359afb795ae9aR6-R9) to a single-line import. Maybe this causes issues in a newer GNOME Shell version.
